packages/svelte/src/compiler/phases/3-transform/client/transform-client.js

```diff
@@ -255,14 +255,24 @@ export function client_component(source, analysis, options) {
 	);
 
 	if (analysis.runes && options.dev) {
-		const bindable = analysis.exports.map(({ name, alias }) => b.literal(alias ?? name));
+		const exports = analysis.exports.map(({ name, alias }) => b.literal(alias ?? name));
+		/** @type {import('estree').Literal[]} */
+		const bindable = [];
 		for (const [name, binding] of properties) {
 			if (binding.kind === 'bindable_prop') {
 				bindable.push(b.literal(binding.prop_alias ?? name));
 			}
 		}
 		instance.body.unshift(
-			b.stmt(b.call('$.validate_prop_bindings', b.id('$$props'), b.array(bindable)))
+			b.stmt(
+				b.call(
+					'$.validate_prop_bindings',
+					b.id('$$props'),
+					b.array(bindable),
+					b.array(exports),
+					b.id(`${analysis.name}.filename`)
```

Comment:
oh actually you know what? we can just pass the function itself through here and use `component.name` instead of reverse engineering it from `filename`. that way it's more likely to match the user's intentions — `foo/index.svelte` becomes `<Foo>` rather than `<index>`

making that change now

Reply:
The name will be `App_1` if there's a variable with the name `App` inside it, so it's not guaranteed to look correct in all cases (but probably fine for 99% of all cases)